pkgs/io_file/lib/src/vm_posix_file_system.dart

```diff
@@ -134,4 +147,58 @@ base class PosixFileSystem extends FileSystem {
       rethrow;
     }
   }
+
+  @override
+  void writeAsBytes(
+    String path,
+    Uint8List data, [
+    WriteMode mode = WriteMode.failExisting,
+  ]) {
+    var flags = stdlibc.O_WRONLY | stdlibc.O_CLOEXEC;
+
+    flags |= switch (mode) {
+      WriteMode.appendExisting => stdlibc.O_CREAT | stdlibc.O_APPEND,
+      WriteMode.failExisting => stdlibc.O_CREAT | stdlibc.O_EXCL,
+      WriteMode.truncateExisting => stdlibc.O_CREAT | stdlibc.O_TRUNC,
+      _ => throw ArgumentError.value(mode, 'invalid write mode'),
+    };
+
+    final fd = _tempFailureRetry(
+      () => stdlibc.open(path, flags: flags, mode: _defaultMode),
```

Comment:
It seems that "mode" (as in `mode` and in `_defaultMode`) is used to represent both flags and permissions(posix "mode"). Would it be possible to use `mode` only for one of those?

Reply:
`dart:io` calls this `FileMode`. The POSIX specification calls it a "flag" and the win32 API calls it `dwCreationDisposition`. How about `WriteOption`?

Reply:
But I might want to revise this when I support generic file open because I'm not sure if you can have seperate read/write options or it you have to fuse them.

Reply:
No strong opinion regarding the name for opening mode(create/append) vs permissions used for new file creation.

Reply:
OK, I'll leave it for now and revisit the naming when I implement the Windows version and add an `open` method.